tests/integration2/galley/conversion/conversion_test.go

```diff
@@ -40,32 +40,33 @@ func TestConversion(t *testing.T) {
 				return
 			}
 
-			// Call Requires to explicitly initialize dependencies that the test needs.
-			ctx := framework.GetContext(t)
-
-			// TODO: Limit to Native environment until the Kubernetes environment is supported in the Galley
-			// component
-			ctx.RequireOrSkip(t, lifecycle.Suite, &descriptors.NativeEnvironment)
-
-			ctx.RequireOrFail(t, lifecycle.Test, &ids.Galley)
-
-			gal := components.GetGalley(ctx, t)
-
-			for i, fset := range d.FileSets() {
-				testName := d.TestName()
-				if len(d.FileSets()) != 1 {
-					runTest(t, fset, gal)
-					testName = fmt.Sprintf("%s_%d", d.TestName(), i)
+			if len(d.FileSets()) == 1 {
+				// Only a single test, just run it using the current test.
+				runTest(t, d.FileSets()[0])
+			} else {
+				// This test is composed of multiple sub-tests. Run each separately.
+				for i, fset := range d.FileSets() {
+					testName := fmt.Sprintf("%s_%d", d.TestName(), i)
+					t.Run(testName, func(t *testing.T) {
+						runTest(t, fset)
+					})
 				}
-				t.Run(testName, func(t *testing.T) {
-					runTest(t, fset, gal)
-				})
 			}
 		})
 	}
 }
 
-func runTest(t *testing.T, fset *testdata.FileSet, gal components.Galley) {
+func runTest(t *testing.T, fset *testdata.FileSet) {
+	// Call Requires to explicitly initialize dependencies that the test needs.
+	ctx := framework.GetContext(t)
+
+	// TODO: Limit to Native environment until the Kubernetes environment is supported in the Galley
+	//  component
+	ctx.RequireOrSkip(t, lifecycle.Suite, &descriptors.NativeEnvironment)
+
+	ctx.RequireOrFail(t, lifecycle.Test, &ids.Galley)
+	gal := components.GetGalley(ctx, t)
```

Comment:
This is not correct. The whole point of the fileset approach is to apply the same set without back-to-back, without resetting the state of the component in-between. This will end up applying the configuration to the resetted version of Galley.

Reply:
@ozevren the ingress tests (the only ones that currently use stages) in their current form seem to be very flaky on osx, which is why I started looking at this code.

FWICT, the "stage" approach is only attempting to reuse the mesh config.  If that's the case, simply duplicating the meshconfig for each stage seems reasonable. Or were you expecting more the state to be carried over? If so, there seems to be a wrinkle in that any call to galley.SetMeshConfig effectively performs a reset ... so if multiple mesh configs were specified for a test sequence, state would be lost at that point.

I suppose I revert these changes and live with the failed tests on osx for now, since you're already doing work to completely refactor the framework (which will give us much better control over component lifecycles).  Sound good?